### PR TITLE
Changing Accuracy to Quality in screed-related files 

### PR DESCRIPTION
**Pull request by bocajnotnef**
See #685

### CODE REVIEW

**Comment by mr-c** [comment]
@bocajnotnef Please bring this branch up to date

**Comment by mr-c** [comment]
- [ ] cut an RC for screed and publish to testpypi; update khmer's setup.py in `refactor/acuqual` to reference that URL
- [ ] get this branch through code review
- [ ] you or I will run the khmer acceptance tests at the point
- [ ] then cut a screed release
- [ ] finally update khmer's setup.py to point to the released version of screed (in a new PR)

**Comment by ctb** [comment]
How's this going?

**Comment by bocajnotnef** [comment]
Am working on it. Working on getting master merged in.

**Comment by bocajnotnef** [comment]
retest this please

**Comment by bocajnotnef** [comment]
restest this please

**Comment by bocajnotnef** [comment]
@mr-c CR, please

File: khmer/thread_utils.py [hunk 1]
```diff
@@ -181,9 +181,9 @@ def do_write(self, outfp):
             except Queue.Empty:
                 continue
 
-            for name, seq, accuracy in g.seqlist:
-                if accuracy:  # write FASTQ; CTB hack.
-                    outfp.write('@%s\n%s\n+\n%s\n' % (name, seq, accuracy))
+            for name, seq, quality in g.seqlist:
+                if quality:  # write FASTQ; CTB hack.
```
Comment (ctb) [inline khmer/thread_utils.py]: could you update this to use write_record, please?

File: sandbox/sweep-reads.py [hunk 2]
```diff
@@ -52,10 +52,10 @@ def fmt_fasta(name, seq, labels=[]):
         name=name, labels='\t'.join([str(l) for l in labels]), seq=seq)
 
 
-def fmt_fastq(name, seq, accuracy, labels=[]):
+def fmt_fastq(name, seq, quality, labels=[]):
```
Comment (ctb) [inline sandbox/sweep-reads.py]: update to use write_record, pls

File: sandbox/sweep-reads.py [hunk 3]
```diff
@@ -284,11 +284,11 @@ def main():
                     ht.consume_sequence_and_tag_with_labels(record.sequence,
                                                             label)
 
-                    if hasattr(record, 'accuracy'):
-                        outfp.write('@{name}\n{seq}+{accuracy}\n'.format(
+                    if hasattr(record, 'quality'):
+                        outfp.write('@{name}\n{seq}+{quality}\n'.format(
```
Comment (ctb) [inline sandbox/sweep-reads.py]: write_record

File: sandbox/trim-low-abund.py [hunk 4]
```diff
@@ -40,17 +40,17 @@ def output_single(read, trim_at=None):
...
     if trim_at is not None:
         sequence = sequence[:trim_at]
-        if accuracy:
-            accuracy = accuracy[:trim_at]
+        if quality:
+            quality = quality[:trim_at]
 
-    if accuracy:
-        return "@%s\n%s\n+\n%s\n" % (name, sequence, accuracy)
+    if quality:
```
Comment (ctb) [inline sandbox/trim-low-abund.py]: Note, this has already been changed to use write_record in another PR (#759) that is waiting for a code review...

File: scripts/extract-paired-reads.py [hunk 5]
```diff
@@ -36,18 +36,18 @@ def is_pair(name1, name2):
 
 
 def output_pair(read1, read2):
-    if hasattr(read1, 'accuracy'):
+    if hasattr(read1, 'quality'):
         return "@%s\n%s\n+\n%s\n@%s\n%s\n+\n%s\n" % \
-            (read1.name, read1.sequence, read1.accuracy,
-             read2.name, read2.sequence, read2.accuracy)
+            (read1.name, read1.sequence, read1.quality,
```
Comment (ctb) [inline scripts/extract-paired-reads.py]: Also updated in #759.

File: scripts/partition-graph.py [hunk 6]
```diff
@@ -43,7 +43,7 @@ def __debug_vm_usage(msg):  # pylint: disable=unused-argument
 
 
 def worker(queue, basename, stop_big_traversals):
-    while 1:
+    while True:
```
Comment (ctb) [inline scripts/partition-graph.py]: Irrelevant change - is this from autopep8??

File: scripts/readstats.py [hunk 7]
```diff
@@ -59,7 +59,7 @@ def main():
 
         try:
             input_iter = screed.open(filename)
-        except (IOError, OSError, EOFError), exc:
+        except (IOError, OSError, EOFError) as exc:
```
Comment (ctb) [inline scripts/readstats.py]: also from autopep8??

File: scripts/sample-reads-randomly.py [hunk 8]
```diff
@@ -74,8 +74,8 @@ def get_parser():
 
 
 def output_single(read):
-    if hasattr(read, 'accuracy'):
-        return "@%s\n%s\n+\n%s\n" % (read.name, read.sequence, read.accuracy)
+    if hasattr(read, 'quality'):
```
Comment (ctb) [inline scripts/sample-reads-randomly.py]: this function is no longer used (and is removed in #759)

File: scripts/split-paired-reads.py [hunk 9]
```diff
@@ -100,7 +100,7 @@ def main():
             if is_fastq:
                 print >> fp_out1, '@%s\n%s\n+\n%s' % (record.name,
                                                       record.sequence,
-                                                      record.accuracy)
+                                                      record.quality)
```
Comment (ctb) [inline scripts/split-paired-reads.py]: Obsoleted in #762, which could also use a review.

**Comment by ctb** [comment]
Overall LGTM, although I think merging #759 and #762 first would simplify things a bit, if we can manage it.  @mr-c shall I merge this PR, or what?

File: scripts/partition-graph.py (hunk 6 above)
Comment (bocajnotnef) [inline scripts/partition-graph.py]: Yes.

File: scripts/readstats.py (hunk 7 above)
Comment (bocajnotnef) [inline scripts/readstats.py]: Yarp.

**Comment by ctb** [comment]
I'm not a big fan of the autopep8 stuff happening in the middle of the PR, but I guess it's in line
with the idea that we fix up files as we go.  So fine.

File: khmer/thread_utils.py (hunk 1 above)
Comment (bocajnotnef) [inline khmer/thread_utils.py]: I can't quite figure out how to pass the individual records to write_record; I'm not sure I understand what the setlist is. I assume a list, yet 

```for item in g.setlist:```
```    write_record(item, outfp)```

doesn't work.

Comment (ctb) [inline khmer/thread_utils.py]: On Tue, Feb 10, 2015 at 08:03:56AM -0800, bocajnotnef wrote:
> > @@ -181,9 +181,9 @@ def do_write(self, outfp):
> >              except Queue.Empty:
> >                  continue
> >  
> > -            for name, seq, accuracy in g.seqlist:
> > -                if accuracy:  # write FASTQ; CTB hack.
> > -                    outfp.write('@%s\n%s\n+\n%s\n' % (name, seq, accuracy))
> > +            for name, seq, quality in g.seqlist:
> > +                if quality:  # write FASTQ; CTB hack.
> 
> I can't quite figure out how to pass the individual records to write_record; I'm not sure I understand what the setlist is. I assume a list, yet 
> 
> ```for item in g.setlist:```
> ```    write_record(item, outfp)```
> 
> doesn't work.

ahh, it's a tuple - so you'd need to construct another version of
write_record that took a tuple.  So never mind, leave it as it is.
This code should go away soon anyway :)

--titus
-- 
C. Titus Brown, ctbrown@ucdavis.edu

File: sandbox/sweep-reads.py (hunk 2 above)
Comment (bocajnotnef) [inline sandbox/sweep-reads.py]: Those don't look like output functions, just formatting functions.

**Comment by bocajnotnef** [comment]
rretest this please

**Comment by bocajnotnef** [comment]
@mr-c @ctb CR?

**Comment by bocajnotnef** [comment]
Or are we still waiting for merges in other PRs?

**Comment by ctb** [comment]
I think we're waiting on either a review/merge from @mr-c, or a merge of #759 by @mr-c and then updating this.

**Comment by ctb** [comment]
Or, well, I guess I don't understand how we're dealing with the screed release, exactly.

**Comment by mr-c** [comment]
@ctb I believe screed is waiting on a run of the khmer acceptance tests before screed v0.8 is released. Screed v1.0 is pending your API review.

**Comment by mr-c** [comment]
@bocajnotnef Okay, khmer builds just fine with the new screed; time to run the acceptance tests
(which may be tricky if any of the eel-pond scripts need conversion as well)

**Comment by mr-c** [comment]
I'm going to merge this as is and we'll update the setup.py file once the new version of screed is out.

File: sandbox/sweep-reads.py (hunk 2 above)
Comment (mr-c) [inline sandbox/sweep-reads.py]: @ctb this script uses a ReadBufferManager which is incompatible with write_record.

**Pull request merged**